testing-modules/junit5-annotations/src/test/java/com/baeldung/junit5/templates/UserIdGeneratorTestInvocationContextProvider.java

```diff
@@ -0,0 +1,87 @@
+package com.baeldung.junit5.templates;
+
+import org.junit.jupiter.api.extension.*;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+
+public class UserIdGeneratorTestInvocationContextProvider implements TestTemplateInvocationContextProvider {
+
+    @Override
+    public boolean supportsTestTemplate(ExtensionContext extensionContext) {
+        return true;
+    }
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(ExtensionContext extensionContext) {
+        boolean featureDisabled = false;
+        boolean featureEnabled = true;
+        return Stream.of(
+            userIdGeneratorTestInFeaturDisabledContext(
+                new UserIdGeneratorTestCase(
+                    "Given feature switch disabled When user name is John Smith Then generated userid is JSmith",
+                    featureDisabled, "John", "Smith", "JSmith")),
+            userIdGeneratorTestInFeaturEnabledContext(
+                new UserIdGeneratorTestCase(
+                    "Given feature switch enabled When user name is John Smith Then generated userid is baelJSmith",
+                    featureEnabled, "John", "Smith", "baelJSmith"))
+        );
+    }
+
+    private TestTemplateInvocationContext userIdGeneratorTestInFeaturDisabledContext(UserIdGeneratorTestCase userIdGeneratorTestCase) {
```

Comment:
Consider shortening these names to featureDisabledContext and featureEnabledContext or something similar.

Reply:
Done